device/simulation/tt_simulation_device.cpp

```diff
@@ -0,0 +1,261 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include "common/logger.hpp"
+#include "common/assert.hpp"
+#include "device/driver_atomics.h"
+#include "device/tt_cluster_descriptor.h"
+
+#include "tt_simulation_device.h"
+#include "uv.h"
+
+tt_SimulationDevice::tt_SimulationDevice(const std::string &sdesc_path) : tt_device(sdesc_path){
+    // log_info(tt::LogEmulationDriver, "Instantiating simulation device");
+    soc_descriptor_per_chip.emplace(0, tt_SocDescriptor(sdesc_path));
+    std::set<chip_id_t> target_devices = {0};
+    
+    // Start VCS simulator in a separate process
+    TT_ASSERT(std::getenv("TT_METAL_VCS"), "TT_METAL_VCS not set, please provide path to the VCS binary");
```

Comment:
same comment as above should the toggles be prefixed with `TT_METAL` or `TT_UMD`?

Reply:
For this, I'll probably rename to something like "TT_REMOTE_EXE" to be more general, since this infra can be used for something like zebu in the future. 

The env var `TT_METAL_SIMULATOR_EN` will change to be not specefic to VCS, and I'm open to any suggestions :D .

Reply:
I suck at naming things so I leave that to you but I just figured we may not want to add `TT_METAL_...` inside of UMD?

Reply:
that makes sense, I'll change the prefix to just `TT_*` or something similar

Reply:
changed to `TT_REMOTE_EXE`